projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/repository/ProjectRepository.java

```diff
@@ -24,9 +24,23 @@ public interface ProjectRepository {
   void add(Project project);
 
   /**
-   * Returns a list of {@link ProjectPreview} entities.
+   * Queries previews with a provided offset and limit that supports pagination.
    *
-   * @return a list of projects
+   * @param offset the offset for the search result to start
+   * @param limit  the maximum number of results that should be returned
+   * @return the results in the provided range
+   * @since 1.0.0
+   */
+  List<ProjectPreview> query(int offset, int limit);
```

Comment:
Should we put the Preview object in the domain? Isn't this something that belongs to the view?